### PR TITLE
Handle query parameters

### PR DESCRIPTION
**Pull request by silverjam**
Hello!  I'm opening this PR to start a discussion around these changes, if this is something you'd like to merge I'm happy to write some tests!

### CODE REVIEW

**Comment by danielSanchezQ** [comment]
@silverjam , somehow the notification was buried among the others so I didn't spot this earlier. This looks promising. I'll leave some comments and if you write some tests would be awesome too.

Thanks for the effort!

File: src/lib.rs [hunk 1]
```diff
@@ -41,7 +41,7 @@ use warp::{Filter, Rejection};
 /// Wrapper around a request data.
 ///
 /// It is the type that holds the request data extracted by the [`extract_request_data_filter`](fn.extract_request_data_filter.html) filter.
-pub type Request = (FullPath, Method, HeaderMap, Bytes);
+pub type Request = (FullPath, Option<String>, Method, HeaderMap, Bytes);
```
Comment (danielSanchezQ) [inline src/lib.rs:44]: Can we abstract this in its own type? Or can we use any other type from the libraries we use already?
Something like:
`pub type QueryParameters = Option<String>`

But I'm pretty sure there should be something already for this.

**Review by danielSanchezQ — CHANGES_REQUESTED** [review]

**Comment by silverjam** [comment]
Hi @danielSanchezQ -- thanks for the feedback-- I'll look in to addressing your feedback in the next few days.

**Comment by danielSanchezQ** [comment]
I check the types in both `warp` and `hyper` and they use either `Option<String>` or `Option<&str>`. For our case since we just pass them from one URL to the other, it should enough. So I think we should just go for something as simple as defining and exposing our own wrapping type `QueryParameter` or `UrlQueryParameter`.

**Comment by silverjam** [comment]
> I check the types in both `warp` and `hyper` and they use either `Option<String>` or `Option<&str>`. For our case since we just pass them from one URL to the other, it should enough. So I think we should just go for something as simple as defining and exposing our own wrapping type `QueryParameter` or `UrlQueryParameter`.

Great, I added a wrapper type for `Option<String>` and expanded the tests to validate the query parameters.

File: src/lib.rs [hunk 2]
```diff
@@ -74,10 +79,18 @@ pub fn reverse_proxy_filter(
         .boxed()
 }
 
+pub fn query_params(
```
Comment (danielSanchezQ) [inline src/lib.rs:82]: Nitpicking here, could we rename this to `query_params_filter`?

**Review by danielSanchezQ — CHANGES_REQUESTED** [review]
Just some tiny change comment.
Everything else looks great. Thanks for the effort ❤️

File: src/lib.rs (hunk 2 above)
Comment (silverjam) [reply src/lib.rs]: Done, added some docs too

**Review by danielSanchezQ — APPROVED** [review]
Awesome work. Thanks for the extra effort @silverjam !!

**Comment by silverjam** [comment]
🎉

**Pull request merged**